charts/alfresco-repository/templates/_helpers-database.tpl

```diff
@@ -1,105 +1,45 @@
-{{- end }}
-{{- if hasPrefix "oracle:thin:@" $jdbc_url }}
-  {{- $ora_url := trimPrefix "oracle:thin:" $jdbc_url }}
-  {{- $ora_url = (mustRegexReplaceAllLiteral "^@(tcps?://)?" $ora_url "oracle://") }}
-  {{- $jdbc_url = $ora_url }}
-{{- end }}
-{{- if hasPrefix "sqlserver://" $jdbc_url }}
-  {{- $jdbc_url = trimPrefix "sqlserver://" $jdbc_url }}
-  {{- $query := $jdbc_url | splitList ";" }}
-  {{- $host := "" }}
-  {{- if and (not (empty (index $query 0))) (not (contains "=" (index $query 0))) }}
-    {{- $host = index $query 0 }}
-    {{- $query = rest $query }}
-  {{- end }}
-  {{- $path := "" }}
-  {{- range $query }}
-    {{- if and (hasPrefix "serverName=" .) (empty $host) }}
-      {{- $host = trimPrefix "serverName=" . }}
-      {{- $_ := mustWithout $query . }}
-    {{- end }}
-    {{- if hasPrefix "databaseName=" . }}
-      {{- $path = trimPrefix "databaseName=" . }}
-      {{- $_ := mustWithout $query . }}
-    {{- end }}
-  {{- end }}
-  {{- $ms_url := printf "sqlserver://%s/%s?%s" $host $path ($query | join "&") }}
-  {{- $jdbc_url = $ms_url }}
-{{- end }}
-{{- $parsed_url := urlParse $jdbc_url }}
-{{- if or (empty $parsed_url.host) (empty $parsed_url.hostname) (empty $parsed_url.scheme) (eq "/" $parsed_url.path) }}
-  {{- fail "The provided JDBC URL cannot be parsed please check or raise a bug." }}
-{{- end }}
-{{- mustToJson (dict "jdbc" $parsed_url) }}
-{{- end -}}
+Render database config as env vars
 
-{{/*
-Compute default ports by RDBMS
-*/}}
-{{- define "alfresco-repository.db.default.port" -}}
-{{- $pg_rdbms := dict "name" "postgresql" "port" 5432 }}
-{{- $my_rdbms := dict "name" "mysql" "port" 3306 }}
-{{- $maria_rdbms := dict "name" "mariadb" "port" 3306 }}
-{{- $ora_rdbms := dict "name" "oracle" "port" 1521 }}
-{{- $ms_rdbms := dict "name" "sqlserver" "port" 1434 }}
-{{- range $rdbms := list $pg_rdbms $my_rdbms $maria_rdbms $ora_rdbms $ms_rdbms }}
-{{- eq $rdbms.name $ | ternary $rdbms.port "" }}
-{{- end }}
-{{- end -}}
+Usage: include "alfresco-repository.db.env" $
 
-{{/*
-Compute default driver by RDBMS
 */}}
-{{- define "alfresco-repository.db.default.driver" -}}
-{{- $pg_rdbms := dict "name" "postgresql" "driver" "org.postgresql.Driver" }}
-{{- $my_rdbms := dict "name" "mysql" "driver" "com.mysql.jdbc.Driver" }}
-{{- $maria_rdbms := dict "name" "mariadb" "driver" "org.mariadb.jdbc.Driver" }}
-{{- $ora_rdbms := dict "name" "oracle" "driver" "oracle.jdbc.OracleDriver" }}
-{{- $ms_rdbms := dict "name" "sqlserver" "driver" "com.microsoft.sqlserver.jdbc.SQLServerDriver" }}
-{{- range $rdbms := list $pg_rdbms $my_rdbms $maria_rdbms $ora_rdbms $ms_rdbms }}
-{{- eq $rdbms.name $ | ternary $rdbms.driver "" }}
+{{- define "alfresco-repository.db.env" -}}
+{{- $dbcmCtx := dict "Values" (dict "nameOverride" "alfresco-database") "Chart" .Chart "Release" .Release }}
+{{- with .Values.configuration.db }}
+{{- $dbcm := coalesce (tpl .existingConfigMap.name $) (include "alfresco-repository.fullname" $dbcmCtx) }}
```

Comment:
yes so the parent chart can have configmap names that are linked to specific releases

Reply:
I am not really excited of using tpl in general because it make things hard to understand when they break (on top of the ugly errors helm already provides out of the box)

what's the problem without it? the user has to provide a different values in case want to do install the same stuff multiple times in the same namespace?

Reply:
reverted the whole tpl thing for now